stdlib/pydoc.pyi

```diff
@@ -42,11 +43,17 @@ class Doc:
     PYTHONDOCS: str
     def document(self, object: object, name: str | None = ..., *args: Any) -> str: ...
     def fail(self, object: object, name: str | None = ..., *args: Any) -> NoReturn: ...
+    @abstractmethod
     def docmodule(self, object: object, name: str | None = ..., *args: Any) -> str: ...
+    @abstractmethod
     def docclass(self, object: object, name: str | None = ..., *args: Any) -> str: ...
+    @abstractmethod
     def docroutine(self, object: object, name: str | None = ..., *args: Any) -> str: ...
+    @abstractmethod
     def docother(self, object: object, name: str | None = ..., *args: Any) -> str: ...
+    @abstractmethod
     def docproperty(self, object: object, name: str | None = ..., *args: Any) -> str: ...
+    @abstractmethod
     def docdata(self, object: object, name: str | None = ..., *args: Any) -> str: ...
```

Comment:
At least in 3.9, these are just aliases of `fail`. Why are they treated differently?

Reply:
The fact that these methods are called freely from `doc.Document`, despite the fact that they all raise `TypeError` (due to being aliases of `fail` in the `Doc` class), and without an `except TypeError` guard, indicates to me that the expectation is that they will be overridden in subclasses: https://github.com/python/cpython/blob/15ee55528e1cbc47ef7e9f64186393eee40a49d9/Lib/pydoc.py#L472-L474

And, indeed, all of these methods are overridden in the two subclasses within `pydoc`: `HTMLDoc` and `TextDoc`.